cmd/zfs_object_agent/zettaobject/src/root_connection.rs

```
@@ -104,10 +104,18 @@ impl RootConnectionState {
             let name = nvl.lookup_string("name")?;
             let object_access = Self::get_object_access(&nvl)?;
 
-            Pool::create(&object_access, name.to_str()?, guid).await;
             let mut response = NvList::new_unique_names();
-            response.insert("Type", "pool create done").unwrap();
-            response.insert("GUID", &guid.0).unwrap();
+            match Pool::create(&object_access, name.to_str()?, guid).await {
+                Ok(_) => {
+                    response.insert("Type", "pool create done").unwrap();
+                    response.insert("GUID", &guid.0).unwrap();
+                }
+                Err(err) => {
+                    error!("pool create failed: {:?}", err);
+                    response.insert("Type", "pool create failed").unwrap();
+                    response.insert("cause", "access-failure").unwrap();
```

Seems like the `cause` is primarily used for the dbgmsg.  Maybe we could return the stringified err, so that we can see if it was access denied, bad endpoint, etc. in the zfs_dbgmsg log.

---

Done.
Here are samples of the cause from zfs_dbgmsg log.
```
1636004659   vdev_object_store.c:1260:agent_reader(): got pool create failed cause="Request ID: None Body: <?xml version="1.0" encoding="UTF-8"?><Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Message><BucketName>cloudburst-data-3</BucketName><RequestId>MYHSSRQC8RKKYXDX</RequestId><HostId>QnW7xB7geASfnoewVnT98I5csM1hgSD4/m9zS7c+wWRre0hZTNBho1GrVUAvi27tHARCOB06iuk=</HostId></Error>"
1636004666   vdev_object_store.c:1260:agent_reader(): got pool create failed cause="Request ID: None Body: <?xml version="1.0" encoding="UTF-8"?><Error><Code>AuthorizationHeaderMalformed</Code><Message>The authorization header is malformed; the region 'us-west-12' is wrong; expecting 'us-west-2'</Message><Region>us-west-2</Region><RequestId>8R829T5VN9ZNEK6Y</RequestId><HostId>SlUPSI2wcE/cy7FZC1ic3ifQfA2PIO46eyHRdOHIFyOE2PeSAyvrqniqHoypYZFXiSX60Mq3yoQ=</HostId></Error>"
1636004706   vdev_object_store.c:1260:agent_reader(): got pool create failed cause="deadline has elapsed"
1636004919   vdev_object_store.c:1260:agent_reader(): got pool create failed cause="Couldn't find AWS credentials in environment, credentials file, or IAM role."
1636005046   vdev_object_store.c:1260:agent_reader(): got pool create failed cause="Request ID: None Body: <?xml version="1.0" encoding="UTF-8"?><Error><Code>InvalidAccessKeyId</Code><Message>The AWS Access Key Id you provided does not exist in our records.</Message><AWSAccessKeyId>INVALIDAWSKEYHEREEEE</AWSAccessKeyId><RequestId>ANQY7CDR4PMYN1T7</RequestId><HostId>1VJ/2FBPPkhJINl1Hr1zvXbyoLZF+bXOj+XdxjnnpCQloDneoD7CRA1MUMImKdqVfIZ1k5FqrHQ=</HostId></Error>"
```